src/test/java/software/amazon/awssdk/crt/test/S3ClientTest.java

```diff
@@ -852,7 +854,15 @@ public void testS3Put() throws IOException {
         skipIfAndroid();
         skipIfNetworkUnavailable();
         Assume.assumeTrue(hasAwsCredentials());
-        testS3PutHelper(false, false, null, false, 16 * 1024 * 1024);
+        testS3PutHelper(false, false, null, false, 16 * 1024 * 1024, false);
+    }
+
+    @Test
+    public void testS3PutWithMD5() throws IOException {
```

Comment:
add a comment explaining how we were especially worried about FIPS and MD5 and that's one of the reasons this test is here